### PR TITLE
Feat/settings/button

### PR DESCRIPTION
**Pull request by francelu**
# Implement settings button in top app bar

## Description
This PR introduces the settings button in the top app bar, integrates it to the profile screen. It closes issue #223.

## Changes
- Implemented settings button in top app bar and its logic. Can't have go back and settings button at the same time.  
- Integrated to Profile Screen top app bar the new settings button.  
- Adapted tests and test tags as necessary.
- Completed navigation actions and main activity as necessary to be able to navigate to settings screen.

#### Added
- `SettingsScreen.kt`: awaiting merge of its UI implementation.

#### Modified
- `MainActivity.kt`: for navigation purposes.
- `C.kt`: for test tag for settings button.
- `NavigationActions.kt`: for navigation to settings screen.
- `TopAppBar.kt`: for the new settings button.
- `ProfileScreen.kt`: to navigate to settings screen.
- `TopAppBarTest.kt`: adapted and completed tests as necessary.
- `ProfileScreenTest.kt`: adapted and completed tests as necessary.

#### Removed
- None.

## Dependencies Added
- None.

## Testing
- Adapted and completed tests as necessary in top app bar and profile screen tests.
- E2E passed on feat/settings/button-e2e.

## Screenshots
image

### CODE REVIEW

File: app/src/test/java/com/android/periodpals/ui/navigation/TopAppBarTest.kt [hunk 1]
```diff
@@ -51,7 +51,24 @@ class TopAppBarTest {
   }
```
Comment (agonzalez-r) [inline app/src/test/java/com/android/periodpals/ui/navigation/TopAppBarTest.kt:51]: for consistency you should add 
`composeTestRule.onNodeWithTag(TopAppBar.SEETINGS_BUTTON).assertDoesNotExist()`
to the other tests like `editButtonIsDisplayed()`, `backButtonIsDisplayed()` and `onlyTitleIsDisplayed()`

File: app/src/main/java/com/android/periodpals/ui/navigation/TopAppBar.kt [hunk 2]
```diff
@@ -59,12 +60,20 @@ fun TopAppBar(
     title: String,
     backButton: Boolean = false,
     onBackButtonClick: (() -> Unit)? = null,
+    settingsButton: Boolean = false,
```
Comment (agonzalez-r) [inline app/src/main/java/com/android/periodpals/ui/navigation/TopAppBar.kt:63]: Missing documentation on added parameters

**Review by taghizadlaura — APPROVED** [review]
## Summary
LGTM :
The PR introduces a settings button in the top app bar on the `ProfileScreen` and integrates navigation to the `SettingsScreen`. It adapts the existing tests and ensures that navigation actions are functional. The implementation partially addresses the requirements for the `SettingsScreen`, pending its UI merge.

## Important

### Code Quality
LGTM :
- The code is readable and follows a consistent structure, especially in terms of integrating the settings button to the top app bar.

### Functionality
LGTM :
- The functionality works as described, enabling navigation to the `SettingsScreen` via the settings button.
- The decision to avoid having both the go-back and settings buttons simultaneously is evident.

### Performance
- No apparent performance issues arise from the changes.

### Security
- No security implications identified in this PR.

## Testing
LGTM
- Unit tests for `TopAppBar` and `ProfileScreen` are updated to include the settings button.
- E2E testing confirms that navigation works as intended, ensuring a smooth user experience.

## Documentation
LGTM :
- The PR clearly documents the changes.
- It’s helpful that navigation logic and test updates are explicitly mentioned.

## Suggestions
- Maybe you could already add a `TopAppBar` to the `SettingScreen` to let users go back to the previous screen (`ProfileScreen`). I know Harrishan is working on the full UI for the `SettingScreen`, but since it’s not merged yet, if we navigate to the `SettingScreen` right now, we’re stuck there. It’d be cool to have a way to go back, even if the UI isn’t fully set up yet.

## Steps before PR approved
- Respond to my suggestions.
- Address Ale's comments (I didn’t repeat what she mentioned, but I agree with her; they are minor nitpicks).

**Review by agonzalez-r — APPROVED** [review]
Good implementation of the settings button. 
- Integration of the setting button to `TopAppBar.kt` is consistent (minor nitpick for the tests), testing is correct
- Integration to `ProfileScreen.kt` is good, added corresponding tests 

LGTM
Just make sure to add the documentation of the added paremeters in `TopAppBar.kt` before merging.

File: app/src/main/java/com/android/periodpals/ui/settings/SettingsScreen.kt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                    vertical = MaterialTheme.dimens.small3,
+                )
+                .verticalScroll(rememberScrollState()),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement =
+            Arrangement.spacedBy(MaterialTheme.dimens.small2, Alignment.CenterVertically),
+    ) {
+      // TODO: delete when implementing the screen
+      Text("Settings Screen", modifier = Modifier.fillMaxSize())
+    }
```
[the remark this answers is not in the document]
Comment (francelu) [reply app/src/main/java/com/android/periodpals/ui/settings/SettingsScreen.kt:39]: Changed ! Could you resolve if you think it's good ? :)

File: app/src/main/java/com/android/periodpals/ui/navigation/TopAppBar.kt (hunk 2 above)
Comment (francelu) [reply app/src/main/java/com/android/periodpals/ui/navigation/TopAppBar.kt:63]: Good you saw, I missed that ! :)

File: app/src/test/java/com/android/periodpals/ui/navigation/TopAppBarTest.kt (hunk 1 above)
Comment (francelu) [reply app/src/test/java/com/android/periodpals/ui/navigation/TopAppBarTest.kt:51]: We typically used `assertIsNotDisplayed()`. I did the necessary changes in the other UI tests. E2E still pass :)  
image

**Pull request merged**